tests/common/port_toggle.py

```diff
@@ -47,24 +55,39 @@ def __check_interface_state(state='up'):
 
     logger.info('toggling ports:\n%s', pprint.pformat(ports))
 
+    cmds_down = []
+    cmds_up = []
     for port in ports:
-        duthost.command('config interface shutdown {}'.format(port))
-        if watch:
-            time.sleep(1)
-            watch_system_status(duthost)
+        cmds_down.append('config interface shutdown {}'.format(port))
+        cmds_up.append('config interface startup {}'.format(port))
+
+    duthost.shell_cmds(cmds=cmds_down)
+    if watch:
+        time.sleep(1)
+
+        # Watch memory status
+        memory_output = duthost.shell("show system-memory")["stdout"]
+        logger.info("Memory Status: %s", memory_output)
+
+        # Watch orchagent CPU utilization
+        orch_cpu = duthost.shell("show processes cpu | grep orchagent | awk '{print $9}'")["stdout"]
+        logger.info("Orchagent CPU Util: %s", orch_cpu)
+
+        # Watch Redis Memory
+        redis_memory = duthost.shell("redis-cli info memory | grep used_memory_human")["stdout"]
+        logger.info("Redis Memory: %s", redis_memory)
 
     # verify all interfaces are down
-    pytest_assert(wait_until(3, 1, __check_interface_state, 'down'),
+    pytest_assert(wait_until(20, 5, __check_interface_state, 'down'),
                   "dut ports {} didn't go down as expected"
-                  .format(list(set(ports).difference(set(duthost.interface_facts(up_ports=ports)['ansible_facts']['ansible_interface_link_down_ports'])))))
+                  .format(list(set(ports).difference(__get_down_ports()))))
 
-    for port in ports:
-        duthost.command('config interface startup {}'.format(port))
+    duthost.shell_cmds(cmds=cmds_up)
```

Comment:
Do we need a ```try-catch``` here? 
If pytest_assert in Line 81 raised an exception, I'm afraid those ports remain down.

Reply:
try catch is not the right way to deal with pytest_assert. I think

Reply:
Agree. I changed the code to explicitly catch exceptions and store a failure flag. Then use pytest_assert to raise exception in case failure flag is True after tried startup all ports.